### PR TITLE
Add musl libc definitions to core/stdc/*

### PR DESCRIPTION
**Pull request by yshui**
This is splitted from #1997.

I think I addressed all the comments. Thanks to all the reviewers.

### CODE REVIEW

File: src/core/stdc/fenv.d [hunk 1]
```diff
@@ -309,6 +309,31 @@ else version( Solaris )
 
     alias int fexcept_t;
 }
+else version( CRuntime_Musl )
+{
+    version (X86_64) {
+        struct fenv_t {
```
Comment (ibuclaw) [inline src/core/stdc/fenv.d]: Nit, can we have curly braces on new lines?

File: src/core/stdc/math.d [hunk 2]
```diff
@@ -601,6 +601,136 @@ else version( CRuntime_Glibc )
...
+    ///
+    int fpclassify(double x)    { return __fpclassify(x);  }
+    ///
+    int fpclassify(real x)
+    {
+        return (real.sizeof == double.sizeof)
+            ? __fpclassify(x)
+            : __fpclassifyl(x);
+    }
+    static uint __FLOAT_BITS(float __f)
```
Comment (ibuclaw) [inline src/core/stdc/math.d]: `static` isn't doing anything here.  Should this instead be `private`?

File: src/core/stdc/math.d [hunk 3]
```diff
@@ -601,6 +601,136 @@ else version( CRuntime_Glibc )
...
+    {
+        union __u_t {
+            float __f;
+            uint __i;
+        }
+        __u_t __u;
+        __u.__f = __f;
+        return __u.__i;
+    }
+    static ulong __DOUBLE_BITS(double __f)
```
Comment (ibuclaw) [inline src/core/stdc/math.d]: Likewise `private`.

File: src/core/stdc/stdio.d [hunk 4]
```diff
@@ -383,6 +401,16 @@ else version( CRuntime_Glibc )
     ///
     alias shared(_IO_FILE) FILE;
 }
+else version( CRuntime_Musl )
+{
+    union fpos_t {
```
Comment (ibuclaw) [inline src/core/stdc/stdio.d]: Curly on next line.

File: src/core/stdc/stdio.d [hunk 5]
```diff
@@ -917,6 +945,20 @@ else version( CRuntime_Bionic )
...
 }
+else version( CRuntime_Musl )
+{
+    // needs tail const
+    extern shared FILE* stdin;
+    ///
+    extern shared FILE* stdout;
+    ///
+    extern shared FILE* stderr;
+    enum {
```
Comment (ibuclaw) [inline src/core/stdc/stdio.d]: Curly on next line.

File: src/core/stdc/stdio.d [hunk 6]
```diff
@@ -917,6 +945,20 @@ else version( CRuntime_Bionic )
...
+        _IOFBF = 0,
+        _IOLBF = 1,
+        _IONBF = 2,
```
Comment (ibuclaw) [inline src/core/stdc/stdio.d]: Add `///` comment for each member.

**Comment by yshui** [comment]
Done

**Pull request merged**